lib/rules/lint-require-valid-alt-text.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const AstNodeInfo = require('../helpers/ast-node-info');
+const Rule = require('./base');
+
+function hasAccessibleChild(node) {
+  return AstNodeInfo.hasChildren(node);
+}
+
+module.exports = class A11yLintAltText extends Rule {
+  logNode({ node, message }) {
+    return this.log({
+      message,
+      line: node.loc && node.loc.start.line,
+      column: node.loc && node.loc.start.column,
+      source: this.sourceForNode(node),
+    });
+  }
+  visitor() {
+    return {
+      ElementNode(node) {
+        const isImg = AstNodeInfo.isImgElement(node);
+        const isObj = AstNodeInfo.isObjectElement(node);
+        const isInput = AstNodeInfo.isInputElement(node);
+        const isArea = AstNodeInfo.isAreaElement(node);
+        const hasAriaHiddenAttribute = AstNodeInfo.hasAttributeValue(node, 'aria-hidden', 'true');
+        if (hasAriaHiddenAttribute) {
+          return;
+        }
+        if (isImg) {
+          const hasAltAttribute = AstNodeInfo.hasAttribute(node, 'alt');
+          const altValue = AstNodeInfo.elementAttributeValue(node, 'alt');
+          const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
+          if (!hasAltAttribute) {
+            this.logNode({
+              message:
+                'img tags must have an alt attribute, either with meaningful text, or an empty string for decorative images.',
+              node,
+            });
+          } else {
+            if (altValue === '') {
+              if (['presentation', 'none'].includes(roleValue)) {
+                return;
+              }
+              this.logNode({
+                message:
+                  'img tags must have an alt attribute, either with meaningful text, or an empty string for decorative images.',
+                node,
+              });
+            }
+          }
+        } else if (isInput) {
+          const isImageInput = AstNodeInfo.hasAttributeValue(node, 'type', 'image');
+          if (!isImageInput) {
+            return;
+          }
+          const hasValidAttributes = AstNodeInfo.hasAnyAttribute(node, [
+            'aria-label',
+            'aria-labelledby',
+            'alt',
+          ]);
+
+          if (!hasValidAttributes) {
+            this.logNode({
+              message:
+                '<input> elements with type="image" must have a text alternative through the `alt`, `aria-label`, or `aria-labelledby` attribute.',
```

Comment:
'`<input>` elements with type="image" are no longer standard use. An `<img>` element should be used instead. However if you must use it, it must have a text alternative through the use of one of the following: the `alt`, `aria-label`, or `aria-labelledby` attribute.'